test/ApplicationTest.php

```diff
@@ -621,4 +625,64 @@ public function testPipingNotInvokableErrorMiddlewareRisesException()
         $this->setExpectedException(InvalidMiddlewareException::class);
         $handler('foo', 'bar', 'baz', 'bat');
     }
+
+    public function invalidRequestExceptions()
+    {
+        return [
+            'invalid file'             => [
+                InvalidArgumentException::class,
+                'Invalid value in files specification',
+            ],
+            'invalid protocol version' => [
+                UnexpectedValueException::class,
+                'Unrecognized protocol version (foo-bar)',
+            ],
+        ];
+    }
+
+    /**
+     * @runInSeparateProcess
+     * @preserveGlobalState disabled
+     * @dataProvider invalidRequestExceptions
+     */
+    public function testRunInvokesFinalHandlerWhenServerRequestFactoryRaisesException(
+        $expectedException,
+        $message
+    ) {
+        // try/catch is necessary in the case that the test fails.
+        // Assertion exceptions raised inside prophecy expectations normally
+        // are fine, but in the context of runInSeparateProcess, these
+        // lead to closure serialization errors. try/catch allows us to
+        // catch those and provide failure assertions.
+        try {
+            $requestFactory = Mockery::mock('alias:' . ServerRequestFactory::class)
+                ->shouldReceive('fromGlobals')
+                ->withNoArgs()
+                ->andThrow($expectedException, $message)
+                ->once()
+                ->getMock();
+
+            $finalHandler = function ($request, $response, $err = null) use ($expectedException, $message) {
+                $this->assertEquals(400, $response->getStatusCode());
```

Comment:
Should we use here const `Fig\Http\Message\StatusCodeInterface::STATUS_BAD_REQUEST`?